app/src/main/java/fr/free/nrw/commons/customselector/ui/screens/ViewImageScreen.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.input.pointer.PointerInputScope
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import fr.free.nrw.commons.customselector.domain.model.Image
+import kotlin.math.abs
+
+@Composable
+fun ViewImageScreen(
+    currentImageIndex: Int,
+    imageList: List<Image>,
+) {
+    var imageScale by remember { mutableFloatStateOf(1f) }
+    var imageOffset by remember { mutableStateOf(Offset.Zero) }
+    var imageSize by remember { mutableStateOf(IntSize.Zero) }
+    var containerSize by remember { mutableStateOf(IntSize.Zero) }
+    val coroutineScope = rememberCoroutineScope()
+
+    LaunchedEffect(imageSize) {
+        println("Image Size : $imageSize")
+    }
+
+    val pagerState = rememberPagerState(initialPage = currentImageIndex) { imageList.size }
+
+    val scrollConnection = object : NestedScrollConnection {
+        override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
+            return if (imageScale > 1f) {
+                // If zoomed in, consume the scroll for panning
+                println("Consuming for panning...")
+                available
+            } else if (
+                source == NestedScrollSource.UserInput && abs(pagerState.currentPageOffsetFraction) > 1e-6
+            ) {
+                println("Handling swipe gestures...")
+                // Handle page swipes only if the image isn't zoomed
+                val delta = available.x
+                val consumed = -pagerState.dispatchRawDelta(-delta)
+                Offset(consumed, 0f)
+            } else {
+                println("Just passing the as it is...")
+                Offset.Zero
+            }
+        }
+    }
+
+    HorizontalPager(
+        state = pagerState,
+        key = { imageList[it].id },
+        pageSpacing = 16.dp
+    ) {
+        BoxWithConstraints(
+            modifier = Modifier
+                .fillMaxSize()
+                .onSizeChanged {
+                    containerSize = it
+                },
+            contentAlignment = Alignment.Center
+        ) {
+//            val state = rememberTransformableState { zoomChange, panChange, _ ->
```

Comment:
Dont forget to remove large commented-out blocks of code 😄

Reply:
Okay, I'll do that once the work is done.